cirq-core/cirq/ops/linear_combinations.py

```diff
@@ -729,3 +732,183 @@ def __format__(self, format_spec: str) -> str:
+                be computed. Qbits absent from self.qubits are acted on by
+                the identity. Defaults to the qubits of the projector_dict.
+
+        Returns:
+            A sparse matrix that is the projection in the specified basis.
+        """
+        return sum(
+            coeff * _projector_string_from_projector_dict(vec).matrix(projector_qids)
+            for vec, coeff in self._linear_dict.items()
+        )
+
+    def expectation_from_state_vector(
+        self,
+        state_vector: np.ndarray,
+        qid_map: Mapping[raw_types.Qid, int],
+    ) -> float:
+        """Expectation of the sum projection from a state vector.
+
+        Projects the state vector onto the sum of projectors and computes the expectation of the
+        measurements.
+
+        Args:
+            state_vector: An array representing a valid state vector.
+            qubit_map: A map from all qubits used in this ProjectorString to the
+                indices of the qubits that `state_vector` is defined over.
+        Returns:
+            The expectation value of the input state.
+        """
+        return sum(
+            coeff
+            * _projector_string_from_projector_dict(vec).expectation_from_state_vector(
+                state_vector, qid_map
+            )
+            for vec, coeff in self._linear_dict.items()
+        )
+
+    def expectation_from_density_matrix(
+        self,
+        state: np.ndarray,
+        qid_map: Mapping[raw_types.Qid, int],
+    ) -> float:
+        """Expectation of the sum of projections from a density matrix.
+
+        Projects the density matrix onto the sum of projectors and computes the expectation of the
+        measurements.
+
+        Args:
+            state: An array representing a valid  density matrix.
+            qubit_map: A map from all qubits used in this ProjectorString to the
+                indices of the qubits that `state_vector` is defined over.
+        Returns:
+            The expectation value of the input state.
+        """
+        return sum(
+            coeff
+            * _projector_string_from_projector_dict(vec).expectation_from_density_matrix(
+                state, qid_map
+            )
+            for vec, coeff in self._linear_dict.items()
+        )
+
+    def __iter__(self):
+        for vec, coeff in self._linear_dict.items():
+            yield _projector_string_from_projector_dict(vec, coeff)
+
+    def __len__(self) -> int:
+        return len(self._linear_dict)
+
+    def __truediv__(self, a: numbers.Complex):
+        return self.__imul__(1 / a)
+
+    def __bool__(self) -> bool:
+        return bool(self._linear_dict)
+
+    def __iadd__(self, other: 'ProjectorSum'):
+        self._linear_dict += other._linear_dict
+        return self
+
+    def __add__(self, other: 'ProjectorSum'):
+        result = self.copy()
+        result += other
+        return result
```

Comment:
Looking at how `__add__` and `__iadd__` are implemented in PauliSum:

```python3
def __iadd__(self, other):
      if isinstance(other, numbers.Complex):
          other = PauliSum.from_pauli_strings([PauliString(coefficient=other)])
      elif isinstance(other, PauliString):
          other = PauliSum.from_pauli_strings([other])
    
      if not isinstance(other, PauliSum):
          return NotImplemented
    
      self._linear_dict += other._linear_dict
      return self
    
def __add__(self, other):
  if not isinstance(other, (numbers.Complex, PauliString, PauliSum)):
      if hasattr(other, 'gate') and isinstance(other.gate, identity.IdentityGate):
          other = PauliString(other)
      else:
          return NotImplemented
  result = self.copy()
  result += other
  return result
```
I think this can still be tightened up to type check for ProjectorString, ProjectorSum, complex numbers and return `NotImplemented` otherwise. I think a similar approach would be worth adopting for `sub`, `mul` etc.

Reply:
I tried to do the exact same thing as for PauliSum as much as possible.

I also took the liberty of adding more tests, hope that's OK.